crates/db_views/src/private_message_view.rs

```diff
@@ -53,17 +53,27 @@ fn queries<'a>() -> Queries<
 
     // If its unread, I only want the ones to me
     if options.unread_only.unwrap_or(false) {
-      query = query
-        .filter(private_message::read.eq(false))
-        .filter(private_message::recipient_id.eq(recipient_id));
+      query = query.filter(private_message::read.eq(false));
+      if let Some(i) = options.creator_id {
+        query = query.filter(private_message::creator_id.eq(i))
+      } else {
+        query = query.filter(private_message::recipient_id.eq(recipient_id));
+      }
     }
     // Otherwise, I want the ALL view to show both sent and received
     else {
       query = query.filter(
         private_message::recipient_id
           .eq(recipient_id)
           .or(private_message::creator_id.eq(recipient_id)),
-      )
+      );
+      if let Some(i) = options.creator_id {
+        query = query.filter(
+          private_message::creator_id
+            .eq(i)
+            .or(private_message::recipient_id.eq(i)),
```

Comment:
I think these filters are wrong now and will return messages where the requesting user isnt involved at all. Would be good to add some test cases to get this right.

Reply:
It wont because of the lines before (65-68), those verify the current user is involved and the ones you highlighter verify the `from` user is involved

Reply:
There are two possible cases where both of those are true current->from or from->current